libs/etterlatte-behandling-model/src/main/kotlin/no/nav/etterlatte/libs/common/behandling/Klage.kt

```diff
@@ -165,9 +165,20 @@ data class Klage(
                     JaNei.NEI -> KlageStatus.FORMKRAV_IKKE_OPPFYLT
                 },
             utfall =
-                when (formkrav.erFormkraveneOppfylt) {
-                    JaNei.JA -> this.utfall
-                    JaNei.NEI -> null
+                if (this.erFormkraveneOppfylt() != formkrav.erFormkraveneOppfylt ||
+                    this.erKlagenFramsattInnenFrist() != formkrav.erKlagenFramsattInnenFrist
+                ) {
+                    null
+                } else {
+                    this.utfall
+                },
+            initieltUtfall =
+                if (this.erFormkraveneOppfylt() != formkrav.erFormkraveneOppfylt ||
+                    this.erKlagenFramsattInnenFrist() != formkrav.erKlagenFramsattInnenFrist
+                ) {
+                    null
+                } else {
+                    this.initieltUtfall
```

Comment:
Tror jeg ville hentet ut spørringen inne i if-en i en forklarende variabel, og så bruke den til å se om den setter utfall / intielt utfall til null.

noe sånt kanskje?
```suggestion
utfall = this.utfall.takeIf(formkravIkkeEndret),
initieltUtfall = this.initieltUtfall.takeIf(formkravIkkeEndret),

```